src/app/modules/ix-forms/components/ix-dynamic-form/ix-dynamic-form-item/ix-dynamic-form-item.component.spec.ts

```diff
@@ -1,89 +1,254 @@
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-ip-input-with-netmask')).toBeHidden();
+    });
+
+    it('explorer', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = explorerSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-explorer')).toBeVisible();
+      expect(spectator.query('ix-explorer')).not.toBeDisabled();
+
+      expect(spectator.query('ix-explorer')).not.toBeHidden();
+      spectator.component.dynamicForm.controls.explorer.disable();
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-explorer')).toBeHidden();
+    });
+
+    it('list', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = listSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-list')).toBeVisible();
+      expect(spectator.query('ix-list')).not.toBeDisabled();
+
+      expect(spectator.query('ix-list')).not.toBeHidden();
+      spectator.component.dynamicForm.controls.list.disable();
+      spectator.detectComponentChanges();
+      expect(spectator.query('ix-list')).toBeHidden();
+    });
+
+    it('dict', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = dictSchema;
+      spectator.detectComponentChanges();
+      expect(spectator.query('.label')).toHaveText('Label Dict');
+
+      expect(spectator.queryAll('ix-dynamic-form-item').length).toEqual(3);
+
+      spectator.queryAll('ix-dynamic-form-item').forEach((item) => {
+        expect(item).not.toBeDisabled();
+        expect(item).not.toBeHidden();
+      });
+
+      spectator.component.dynamicForm.controls.dict.disable();
+      spectator.detectComponentChanges();
+
+      spectator.queryAll('ix-dynamic-form-item').forEach((item) => {
+        expect(item).not.toBeDisabled();
+        expect(item).toBeHidden();
+      });
+    });
   });
 
-  describe('Control Emit', () => {
+  describe('Component methods', () => {
+    it('isHidden()', () => {
+      dynamicForm.controls.list.disable();
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = listSchema;
+      expect(spectator.component.isHidden).toBeTruthy();
+    });
+
+    it('getFormArray()', () => {
+      spectator.component.dynamicForm = dynamicForm;
+      spectator.component.dynamicSchema = listSchema;
+      expect(spectator.component.getFormArray).toEqual(dynamicForm.controls.list);
+    });
+
     it('addControl()', () => {
       spectator.component.dynamicForm = dynamicForm;
-      spectator.component.dynamicSchema = dynamicSchema;
+      spectator.component.dynamicSchema = listSchema;
 
       jest.spyOn(spectator.component.addListItem, 'emit').mockImplementation();
       spectator.component.addControl();
       expect(spectator.component.addListItem.emit).toHaveBeenCalledTimes(1);
     });
```

Comment:
Here too `addControl` is not mean to be called directly by anything but the template. It's only public de jure.
What really happens is: user presses add in ix-list and we want to emit `addListItem`.
We can simulate this:
```
    it('emits addListItem when (add) is emitted by a list control', () => {
      spectator = createComponent({
        props: {
          dynamicForm,
          dynamicSchema: listSchema,
        },
      });
      jest.spyOn(spectator.component.addListItem, 'emit').mockImplementation();

      spectator.query(IxListComponent).add.emit();

      expect(spectator.component.addListItem.emit).toHaveBeenCalledWith({
        array: dynamicForm.controls.list,
        schema: listSchema.itemsSchema, // itemsSchema needs to be added
      });
    });
```